File: Stronghold.Core/Entities/StaffMember.cs
using Stronghold.Core.Enums;

namespace Stronghold.Core.Entities;

/// <summary>
/// Jedna tabela za trenere i nutricioniste (StaffType).
/// Radno vrijeme (WorkStartHour/WorkEndHour) je osnova za slobodne satnice termina.
/// </summary>
public class StaffMember : BaseEntity
{
    public string FirstName { get; set; } = null!;
    public string LastName { get; set; } = null!;
    public StaffType StaffType { get; set; }
    public byte[]? ImageData { get; set; }
    public string Biography { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string Phone { get; set; } = null!;
    public int WorkStartHour { get; set; }
    public int WorkEndHour { get; set; }

    public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
}

File: UI/stronghold_desktop/lib/models/staff_member.dart
class StaffMember {
  final int id;
  final String firstName;
  final String lastName;
  final String staffType;
  final String biography;
  final String email;
  final String phone;
  final int workStartHour;
  final int workEndHour;
  final bool hasImage;

  StaffMember({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.staffType,
    required this.biography,
    required this.email,
    required this.phone,
    required this.workStartHour,
    required this.workEndHour,
    required this.hasImage,
  });

  String get fullName => '$firstName $lastName';

  factory StaffMember.fromJson(Map<String, dynamic> json) => StaffMember(
        id: json['id'] as int,
        firstName: json['firstName'] as String,
        lastName: json['lastName'] as String,
        staffType: json['staffType'] as String,
        biography: json['biography'] as String,
        email: json['email'] as String,
        phone: json['phone'] as String,
        workStartHour: json['workStartHour'] as int,
        workEndHour: json['workEndHour'] as int,
        hasImage: json['hasImage'] as bool,
      );
}
